CMakeLists.txt

```diff
@@ -90,86 +85,53 @@ option(IREE_ENABLE_EMITC "Enables MLIR EmitC dependencies." ${IREE_BUILD_COMPILE
-    set(IREE_TARGET_BACKENDS_TO_BUILD ${IREE_ALL_TARGET_BACKENDS})
-
-    # Disable the WebGPU backend by default, as it has complex dependencies and
-    # is still early in its development.
-    list(REMOVE_ITEM IREE_TARGET_BACKENDS_TO_BUILD WebGPU)
-  endif()
-  message(STATUS "Building compiler target backends: ${IREE_TARGET_BACKENDS_TO_BUILD}")
+# Vulkan is not natively supported on Apple platforms.
+# Metal should generally be used instead, though MoltenVK may also work.
+if (APPLE)
+  set(IREE_HAL_DRIVER_VULKAN_DEFAULT OFF)
 else()
-  set(IREE_TARGET_BACKENDS_TO_BUILD "" CACHE STRING "" FORCE)
-  message(STATUS "Compiler is disabled, building no target backends")
+  set(IREE_HAL_DRIVER_VULKAN_DEFAULT ON)
 endif()
 
-# Default every IREE_TARGET_BACKEND_* to OFF
-foreach(_backend ${IREE_ALL_TARGET_BACKENDS})
-  string(TOUPPER "${_backend}" uppercase_backend)
-  set(IREE_TARGET_BACKEND_${uppercase_backend} OFF CACHE BOOL "" FORCE)
-endforeach()
-
-# Set IREE_TARGET_BACKEND_* based on configuration
-foreach(_backend ${IREE_TARGET_BACKENDS_TO_BUILD})
-  string(TOUPPER "${_backend}" uppercase_backend)
-  string(REPLACE "\"" "" uppercase_backend ${uppercase_backend})
-  set(IREE_TARGET_BACKEND_${uppercase_backend} ON CACHE BOOL "" FORCE)
-endforeach()
-
-list(APPEND CMAKE_MODULE_PATH
-  ${CMAKE_CURRENT_LIST_DIR}/build_tools/cmake/
-  ${CMAKE_CURRENT_LIST_DIR}/bindings/python/build_tools/cmake/
-)
+option(IREE_HAL_DRIVER_CUDA "Enables the CUDA runtime HAL driver" ${IREE_HAL_DRIVER_CUDA_DEFAULT})
+option(IREE_HAL_DRIVER_DYLIB "Enables the Dylib runtime HAL driver" ON)
+option(IREE_HAL_DRIVER_DYLIB_SYNC "Enables the Dylib_Sync runtime HAL driver" ON)
+option(IREE_HAL_DRIVER_VMVX "Enables the VMVX runtime HAL driver" ON)
+option(IREE_HAL_DRIVER_VMVX_SYNC "Enables the VMVX_Sync runtime HAL driver" ON)
+option(IREE_HAL_DRIVER_VULKAN "Enables the Vulkan runtime HAL driver" ${IREE_HAL_DRIVER_VULKAN_DEFAULT})
+
+option(IREE_TARGET_BACKEND_CUDA "Enables the CUDA compiler target backend" ${IREE_BUILD_COMPILER})
+option(IREE_TARGET_BACKEND_DYLIB-LLVM-AOT "Enables the Dylib-LLVM-AOT compiler target backend" ${IREE_BUILD_COMPILER})
+option(IREE_TARGET_BACKEND_WASM-LLVM-AOT "Enables the WASM-LLVM-AOT compiler target backend" ${IREE_BUILD_COMPILER})
+option(IREE_TARGET_BACKEND_METAL-SPIRV "Enables the Metal-SPIRV compiler target backend" ${IREE_BUILD_COMPILER})
+option(IREE_TARGET_BACKEND_ROCM "Enables the ROCm compiler target backend" ${IREE_BUILD_COMPILER})
+option(IREE_TARGET_BACKEND_VULKAN-SPIRV "Enables the Vulkan-SPIRV compiler target backend" ${IREE_BUILD_COMPILER})
+option(IREE_TARGET_BACKEND_VMVX "Enables the VMVX compiler target backend" ${IREE_BUILD_COMPILER})
+# Disable WebGPU by default - it has complex deps and is under development.
+option(IREE_TARGET_BACKEND_WEBGPU "Enables the WebGPU compiler target backend" OFF)
+
+message(STATUS "IREE build runtime HAL driver 'CUDA': ${IREE_HAL_DRIVER_CUDA}")
```

Comment:
This is a lot of repetition. Could we have a little macro/function that does both the option creation and the status message? Would still definitely want the variable names as literals. Something like:

```cmake
function(OPTION NAME DEFAULT)
  option(${OPTION} "Enables the '${NAME}' compiler target backend")
  message(STATUS "IREE build compiler target backend '${NAME}': ${OPTION}")
endfunction()
```

Reply:
I actually like the repetition here, as a user of the project. No logic to follow, just WYSIWYG :P

Reply:
Elaborating on this a bit:

When integrating a project that uses CMake, the root `CMakeLists.txt` is the first (and sometimes only) place to look. It is very helpful if that file is organized into sections and has easy to follow logic. I don't think wrapping this in a function offers much benefit for maintainers (find/replace and multicursor editing within 40 contiguous lines of code is easy), and having each option spelled out with no loops, branches, or string manipulation makes it easy for users to scan the file and see which options they may want to enable/disable for their use.